### PR TITLE
Get latest reconciliation statement

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1361

### Change description ###

Get latest recon statement for that day.
Next PRs will make request to processor and other services.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java [hunk 1]
```diff
@@ -62,4 +63,19 @@ public UUID save(NewEnvelopeSupplierStatement statement) throws SQLException {
             return Optional.empty();
         }
     }
+
+    public Optional<EnvelopeSupplierStatement> findLatestCreated(LocalDate date) {
+        try {
+            EnvelopeSupplierStatement statement = jdbcTemplate.queryForObject(
+                "SELECT * FROM envelope_supplier_statements WHERE DATE(created_at) = :date "
+                    + "order by created_at desc "
+                    + "limit 1",
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java:72]: would be nice to keep letters capital for reserved sql words ;)

**Review by doncem — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java [hunk 2]
```diff
@@ -62,4 +63,19 @@ public UUID save(NewEnvelopeSupplierStatement statement) throws SQLException {
             return Optional.empty();
         }
     }
+
+    public Optional<EnvelopeSupplierStatement> findLatestCreated(LocalDate date) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/statements/SupplierStatementRepository.java:67]: `findLatest` is clean enough I think

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java [hunk 3]
```diff
@@ -23,9 +26,16 @@
 @SpringBootTest
 public class SupplierStatementRepositoryTest {
 
-    @Autowired SupplierStatementRepository repo;
-    @Autowired ObjectMapper objectMapper;
-    @Autowired ClockProvider clockProvider;
+    @Autowired private SupplierStatementRepository repo;
+    @Autowired private ClockProvider clockProvider;
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java:30]: is this change needed (adding `private`)?

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java [hunk 4]
```diff
@@ -87,4 +97,45 @@ void should_return_empty_optional_if_statement_does_not_exist() {
...
+                statementDate,
+                "{ \"b\": \"300\" }",
+                "v1.0.0"
+            );
+
+        // when
+        UUID id = repo.save(newEnvelopeSupplierStatement);
+        assertThat(id).isNotNull();
+
+        var latestAfterThisTime = LocalDateTime.now(clockProvider.getClock());
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java:127]: why is this one using clock and the other one (line 115) is not?

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java (hunk 3 above)
Comment (mdayican) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/SupplierStatementRepositoryTest.java:30]: not needed. just to be consistent.

**Review by rkondratowicz — APPROVED** [review]
